### DISCUSSION TITLE
Statamic form with multiple file upload using Filepond

### DISCUSSION THREAD

**Discussion by laborb-mp** [statamic/cms#6604] [Q&A]
Hey there,

I'm trying to upload multiple files with Filepond (https://pqina.nl/filepond/).
I've created a Statamic form with an Assets Field (handle name `files`) and replaced the default and classic input (type: file) with Filepond (see: https://pqina.nl/filepond/docs/getting-started/examples/replace-file-input/). Worked like a charme and the uploaded files are stored in a disk/asset driver `assets:applications`.

Now I'm trying to upload my files asynchronosly (see: https://pqina.nl/filepond/docs/getting-started/examples/async-upload/). 
I've added the CSRF Token, but whenever I submit the whole form the files aren't stored although Filepond returns no errors and it seems that Filepond uploaded the files without problems.

That's the code:

```
create(input, {
    // Validate accepted file types
    acceptedFileTypes: ['application/pdf', 'image/*', 'zip', 'application/zip'],
    // Validate filesize
    maxFileSize: '20MB',
    maxTotalFileSize: '20MB',
    // Replace classic input field w/ Filepond dropzone
        // storeAsFile: true,
    // Activate asyn. uploading
    server: {
        url: 'https://my-site-url.ddev.site/!/forms/application_form',
        headers: {'X-CSRF-Token': 'IrBx232xuVI1d12e1YeeID6bd2F2gkQIRibk9wDoT23wAVlW2Lef982y'},
    },
});
```
and

```
{{ form:application_form files="true" }}
    {{ fields }}
        <div>
          {{ field }}
          <!-- ... -->
            <input type="file" name="files[]" multiple="" required="">
          <!-- ... -->
        </div>
    {{ /fields }}

    <div>
        <button type="submit">
            Submit
        </button>
    </div>
{{ /form:application_form }}
```

Do you have any ideas what I'm dong wrong?
Thanks in advance!

**Comment by duncanmcclean** [comment]
You say the files aren't stored and Filepond gives you a success message - is the rest of the form submitted or not?

I implemented Filepond back in the v2 days on a project and I created a separate endpoint for the files to upload to. From that, I then had the path that could be used when submitting the form.
